R/pool-methods.R

```diff
@@ -0,0 +1,89 @@
+#' @include pool.R
+NULL
+
+#' S4 class for compatibility with DBI methods
+#' @export
+setClass("Pool")
+
+## documented manually, with the Pool object
+#' @export
+poolCreate <- function(factory, ...,
+                       minSize = 1, maxSize = Inf,
+                       idleTimeout = 60000, validateTimeout = 2000,
+                       stateEnv = NULL) {
```

Comment:
Right. As I mentioned above, I was still picturing `dbPool` and `poolCreate` as equivalents until now (i.e. you could pass the same arguments to other and the other directly). Without that restriction anymore, this does feel neater.

It's funny how, with these changes, `poolCreate` has just become a very think wrapper around `Pool$new`... Do you think we even need `poolCreate` at this point...? Especially if we're not going to tell normal users about it (only future backend authors), it seems likes it isn't useful at all (except maybe to keep the symmetry with `poolClose`, but maybe that's not a good enough reason to keep it around...)